docs/book/src/architecture/controllers/machine.md

```diff
@@ -1 +1,86 @@
-# Machine
+# Machine  Controller
+
+The Machine controller's main responsibilities are:
+
+* Setting an OwnerReference on:
+    * Each Machine object to the Cluster object.
+    * The associated BootstrapConfig object.
+    * The associated InfrastructureMachine object.
+* Setting NodeRefs to be able to associate machines and kubernetes nodes.
+* Cleanup of related objects.
+* Keeping the Machine's Status object up to date with the InfrastructureMachine's Status object.
+
+## Contracts
+
+### Cluster API
+
+Cluster associations are made via labels.
+
+A machine must have a label with a key of `cluster.x-k8s.io/cluster-name` and a value of the core Cluster's name.
+
+### Bootstrap provider
```

Comment:
Should there also be mention here that if a user (or external system) provides Machine.Spec.Bootstrap.Data that the bootstrap provider and any related config will be unused?

Reply:
Yes, that's a reasonable behavior to document!